src/Infrastructure/Scout/Builder.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JeroenG\Explorer\Infrastructure\Scout;
+
+use JeroenG\Explorer\Application\Paginator;
+use JeroenG\Explorer\Domain\Aggregations\AggregationSyntaxInterface;
+use JeroenG\Explorer\Domain\Query\QueryProperties\QueryProperty;
+use JeroenG\Explorer\Domain\Syntax\Compound\BoolQuery;
+
+class Builder extends \Laravel\Scout\Builder
```

Comment:
Hmm. I would like to see this either marked as a final class. I don't want to maintain this as a public API in the package.

Reply:
ahhh shame, this could be a bit of a issue on my side. As a package user, I'm extending Builder as well to add the join feature and override pagination behaviour. If the class becomes final I would need to duplicate or decorate this Builder and rewrite any of the existing methods. 

Any ideas that could work for the both of us? Ultimately this is your package so I don't want to add anything that goes against your vision.

Reply:
https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/internal.html

This might be a compromise, it'll make extending it your own risk. :)

Reply:
For the time being, I've added the internal class tag but I do like the idea of extending a CommandBuilder and then in other cases defaulting back to the scout wrapper. 

I know that `elastic-scout-driver-plus` handles it by adding a searchQuery() method to Searchable and returns a common SearchRequestBuilder class for any advanced querying. This way there's a clear distinction between typical scout behaviour and any advanced use cases.